examples/complete/data.tf

```diff
@@ -0,0 +1,122 @@
+data "aws_partition" "current" {}
+
+data "aws_region" "current" {}
+
+data "aws_organizations_organization" "org" {}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "s3" {
+  version = "2012-10-17"
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    actions   = ["s3:GetEncryptionConfiguration"]
+    resources = ["arn:${local.partition}:s3:::${lower(var.name)}"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "${local.partition}:PrincipalOrgID"
+      values   = [data.aws_organizations_organization.org.id]
+    }
+  }
+
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    actions = [
+      "s3:PutObject",
+      "s3:PutObjectAcl"
+    ]
+    resources = ["arn:${local.partition}:s3:::${lower(var.name)}/*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "${local.partition}:PrincipalOrgID"
+      values   = [data.aws_organizations_organization.org.id]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "kms_policy" {
+  #checkov:skip=CKV_AWS_109: "Ensure IAM policies does not allow permissions management / resource exposure without constraints"
+  #checkov:skip=CKV_AWS_356: "Ensure no IAM policies documents allow "*" as a statement's resource for restrictable actions"
+  #checkov:skip=CKV_AWS_111: "Ensure IAM policies does not allow write access without constraints"
+  statement {
+    sid    = "Allow Session Manager encrypt the session"
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
```

Comment:
Since you have down bellow a statement for the root users do you still need this open statement? Seems redundant.

Reply:
Removed